erpnext/maintenance/doctype/maintenance_visit/maintenance_visit.py

```diff
@@ -16,8 +17,34 @@ def validate_serial_no(self):
 			if d.serial_no and not frappe.db.exists("Serial No", d.serial_no):
 				frappe.throw(_("Serial No {0} does not exist").format(d.serial_no))
 
+	def validate_mntc_date(self):
+		if self.maintenance_type == "Scheduled":
+			p = self.purposes
+			for i in p:
+				detail_ref = i.prevdoc_detail_docname
+			item_ref = frappe.db.get_value('Maintenance Schedule Detail', detail_ref , 'item_ref')
+			start_date, end_date = frappe.db.get_value('Maintenance Schedule Item', item_ref, ['start_date', 'end_date'])
```

Comment:
- add an extra row in maintenance visit
- this row ofcourse wont have any link to the maintenance schedule
- save
![Screenshot 2021-05-26 at 1 30 39 PM](https://user-images.githubusercontent.com/25857446/119623788-255fc480-be26-11eb-8de3-a71956e10468.png)

`start_date` and `end_date` will be blank as the last row returns None for the value of `detail_ref`

Mapping design can be better. 
Map the the Maintenance Schedule Row in the parent form itself? Since we have a one to one mapping between Maintenance Schedule Row and Maintenance Visit.
If the details dont exist on the parent dont validate, its a stray Maintenance Visit, not created from a schedule.
Make the link details hidden